src/js/components/sidebar.jsx

```diff
@@ -44,6 +45,9 @@ export default class SideBar extends React.Component {
     browser.tabs.onMoved.addListener(this._onTabMoved);
     browser.tabs.onRemoved.addListener(this._onTabRemoved);
     browser.tabs.onUpdated.addListener(this._onTabUpdated);
+    browser.tabs.onDetached.addListener(this._onTabDetached.bind(this));
```

Comment:
You'll notice that on lines 30 -- 35 we replace the methods with the methods bound to the current context. We do this so that when we unmount the component in `componentWillUnmount` we can unattach the listener. 

`this._onTabDetached.bind(this)` will return an anonymous function so doing
`browser.tabs.onDetached.removeListener(this._onTabDetached)` wont remove the listener.

Replace the method on the object with it bound to the context. Same below.

You will also need to unbind them in `componentWillUnmount`